python/raft-ann-bench/src/raft-ann-bench/run/conf/wiki_all.json

```diff
@@ -0,0 +1,248 @@
+        { "nprobe": 200, "internalDistanceDtype": "half", "smemLutDtype": "half", "refine_ratio": 32 },
+        { "nprobe": 500, "internalDistanceDtype": "half", "smemLutDtype": "half", "refine_ratio": 32 }
+      ]
+    },
+    {
+      "name": "raft_ivf_pq.d32X-nlist16K",
+      "algo": "raft_ivf_pq",
+      "build_param": {
+        "pq_dim": 32,
+        "pq_bits": 8,
+        "nlist": 16384,
+        "niter": 10,
+        "ratio": 10
+      },
+      "file": "wiki_all/raft_ivf_pq/d32-nlist16K",
+      "search_params": [
+        { "nprobe": 20, "internalDistanceDtype": "half", "smemLutDtype": "half", "refine_ratio": 16 },
+        { "nprobe": 30, "internalDistanceDtype": "half", "smemLutDtype": "half", "refine_ratio": 16 },
+        { "nprobe": 40, "internalDistanceDtype": "half", "smemLutDtype": "half", "refine_ratio": 16 },
+        { "nprobe": 50, "internalDistanceDtype": "half", "smemLutDtype": "half", "refine_ratio": 16 },
+        { "nprobe": 100, "internalDistanceDtype": "half", "smemLutDtype": "half", "refine_ratio": 16 },
+        { "nprobe": 200, "internalDistanceDtype": "half", "smemLutDtype": "half", "refine_ratio": 16 },
+        { "nprobe": 500, "internalDistanceDtype": "half", "smemLutDtype": "half", "refine_ratio": 16 },
+        { "nprobe": 30, "internalDistanceDtype": "half", "smemLutDtype": "half", "refine_ratio": 8 },
+        { "nprobe": 40, "internalDistanceDtype": "half", "smemLutDtype": "half", "refine_ratio": 8 },
+        { "nprobe": 50, "internalDistanceDtype": "half", "smemLutDtype": "half", "refine_ratio": 8 },
+        { "nprobe": 100, "internalDistanceDtype": "half", "smemLutDtype": "half", "refine_ratio": 8 },
+        { "nprobe": 200, "internalDistanceDtype": "half", "smemLutDtype": "half", "refine_ratio": 8 },
+        { "nprobe": 500, "internalDistanceDtype": "half", "smemLutDtype": "half", "refine_ratio": 8 },
+        { "nprobe": 30, "internalDistanceDtype": "half", "smemLutDtype": "half", "refine_ratio": 4 },
+        { "nprobe": 40, "internalDistanceDtype": "half", "smemLutDtype": "half", "refine_ratio": 4 },
+        { "nprobe": 50, "internalDistanceDtype": "half", "smemLutDtype": "half", "refine_ratio": 4 },
+        { "nprobe": 100, "internalDistanceDtype": "half", "smemLutDtype": "half", "refine_ratio": 4 },
+        { "nprobe": 200, "internalDistanceDtype": "half", "smemLutDtype": "half", "refine_ratio": 4 },
+        { "nprobe": 500, "internalDistanceDtype": "half", "smemLutDtype": "half", "refine_ratio": 4 }
+
+      ]
+    },
+    {
+      "name": "raft_cagra.dim32.single_cta",
+      "algo": "raft_cagra",
+      "build_param": {
+        "graph_degree": 32,
+        "intermediate_graph_degree": 48 ,
+        "ivf_pq_build_params": {
+          "pq_dim": 32,
+          "pq_bits": 8,
+          "nlist": 16384,
+          "niter": 10,
+          "ratio": 10
+        },
+      "ivf_pq_search_params": { "nprobe": 10, "internalDistanceDtype": "half", "smemLutDtype": "half", "refine_ratio": 4 }},
```

Comment:
Currently both `ivf_pq_build_params` and `ivf_pq_search_params` are part of CAGRA's `build_param`. 

I do not see how flattening these would change anything in our ability to do grid search. 

I think keeping the current structure is easier to read: parameters are grouped according to their function, and names are shorter.

Btw, you probably do not want to run a full grid search over `graph_degree` and full combination of `ivf_pq_build_params` and `ivf_pq_search_params`. The right approach is to run an `ivf_pq` grid search to find reasonable build / search config, and then just copy the best candidates over here. Keeping the same structure makes it simpler do so so.

Having said that, if you are convinced that the flattened names are better, then we can change them.

Reply:
One more thing: I will delete this file, and add the new cagra configs to `wiki_all_1M` / `wiki_all_10M` and `wiki_all_88M.json` files.

Reply:
We need to expose these properties through the yaml config files and we can't do that today because we cant specify nested objects as arguments to the build/search params. We can, however, specify simple objects in flattened keys (and even just lock them to a single value instead of specifying a range). The nice thing about this is that a range could be specified easily if so desired. 

We're pretty much past specifying this stuff manually in json files at this point. At a minimum, we need to make sure we can specify these things in the yaml files.

Reply:
Even developing this PR is includes manually tweaking json files. But I agree that we need to specify these params at a higher level as well.

I agree that we should keep things flat in the yaml config. So the question is: what is the simplest way to generate the json config from yaml? The python wrapper for the benchmark is not aware of special needs of individual algorithms, so it is indeed the C++ layer which can map the parameters into the correct structure the easiest way. I have modfied the PR accordingly.